[ADD] Module report_controller

hmrodrigues opened the pull request:
Reimplemented the controller (pdf) report type, missing since Odoo 11.0
It duplicates some upstream code, since upstream has the base types hardcoded,
if the PR https://github.com/odoo/odoo/pull/28365 is merged and backported
into 12.0, we will need to refactor in order to remove the duplicated
code.

hmrodrigues committed:
[ADD] Module report_controller

Reimplemented the controller (pdf) report type, missing since Odoo 11.0
It duplicates some upstream code, since upstream has the base types hardcoded,
if the PR https://github.com/odoo/odoo/pull/28365 is merged and backported
into 12.0, we will need to refactor in order to remove the duplicated
code.

pedrobaeza commented:
I don't get the goal of this module. Why do you want to reimplement it? What do you gain with this? All of these questions should be included in the README.

hmrodrigues commented:
> I don't get the goal of this module. Why do you want to reimplement it? What do you gain with this? All of these questions should be included in the README.

Hello Pedro, the goal of this module is to print reports based on data from a external service.

For example, in Portugal, all customer invoices must be certified and the PDF must be printed by the certified software. Since only the product owners can certify their software, in Odoo case is Odoo SA, the certification software is usually external to Odoo, making the Odoo report legally invalid. To print the external PDF using the print menu, we could to call a controller to fetch the PDF from the external service.

hmrodrigues committed:
[IMP] README

Added module goal to README

hmrodrigues committed:
[FIX] lint issues reported by CI

hmrodrigues committed:
[FIX] More lint fixes

hmrodrigues committed:
[IMP] Improved code quality reported by CI

pedrobaeza commented:
OK, thanks for the explanation.

Commit:
[FIX] fixed require location

AWL-TKO committed:
[ADD] Report Controller Extensions

 - Report Session Cookie Persistence
 - Allows to add other methods of identifying controller url
 - Auto add host if necessary
 - Allow other content type for report request

AWL-TKO committed:
[ADD] Improved Cookie Salvaging

Allows to dynamically salvage cookie parameters
Added new error message for error responses

AWL-TKO committed:
[ADD] 403

403 message
